lib/semian/net_http.rb

```diff
@@ -67,9 +76,11 @@ def reset_exceptions
     Semian::NetHTTP.reset_exceptions
 
     def raw_semian_options
-      options = semian_configuration
-      options = options.dup unless options.nil?
-      options
+      check_semian_configuration_changed
+      @raw_semian_options ||= begin
+        @raw_semian_options = Semian::NetHTTP.retrieve_semian_configuration(address, port)
+        @raw_semian_options = @raw_semian_options.dup unless @raw_semian_options.nil?
+      end
```

Comment:
I think my main point here is that it is awkward to argue for resetting the cached `@resource` based on condition `@configuration_id != Semian::NetHTTP.semian_configuration.object_id` but not have a similar cache for `@raw_semian_options`